torch/_guards.py

```diff
@@ -148,10 +148,6 @@ class Guard:
     stack = None
     user_stack = None
 
-    def __post_init__(self):
-        self.stack = CapturedTraceback.extract(skip=2)
-        self.user_stack = TracingContext.extract_stack()
```

Comment:
@jansel told me and I agree that the construction site of guards is not as useful as when they are installed.

Reply:
Why not both? Both have value imo.

The way I wanted to do guard creation logging is to add a `reason` field - most initial guard association/creation sites are manual/have some reason that we hold in tribal knowledge.

Reply:
It's a signal-to-noise problem. Yes, I *could* report the allocation site and the installation site but you're doubling the amount of information you have to sift through. A lot of the trickiness here is making a good decision about what to give to the user by default. That being said, I'm amenable to changing up what we record, it's not a hard change, we can tweak it as we get experience.